lib/join/env0/validator.go

```diff
@@ -0,0 +1,80 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package env0
+
+import (
+	"context"
+
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+	"github.com/zitadel/oidc/v3/pkg/client/rp"
+
+	"github.com/gravitational/teleport/lib/oidc"
+)
+
+const (
+	// env0IssuerURL is the env0 public issuer URL
+	env0IssuerURL = "https://login.app.env0.com/"
+
+	// env0Audience is the audience for the token. This is unfortunately hard
+	// coded.
+	env0Audience = "https://prod.env0.com"
```

Comment:
The example token in their public docs unfortunately specify the wrong value here 🙁  https://docs.envzero.com/guides/integrations/oidc-integrations#format-of-the-openid-connect-id-token

(At least the value in the table is correct, but I'm glad I was able to test it explicitly...)